src/Microsoft.AspNet.Http.Abstractions/PathString.cs

```diff
@@ -66,7 +67,24 @@ public override string ToString()
         public string ToUriComponent()
         {
             // TODO: Measure the cost of this escaping and consider optimizing.
-            return HasValue ? string.Join("/", _value.Split('/').Select(UrlEncoder.Default.UrlEncode)) : string.Empty;
+            if (!HasValue)
+            {
+                return string.Empty;
+            }
+            var values = _value.Split(splitChar);
```

Comment:
Nice! This makes me feel like we should make a non allocating string split using https://github.com/aspnet/Common/blob/dev/src/Microsoft.Extensions.Primitives/StringSegment.cs